### ISSUE TITLE
Activation script alter $IFS, hence altering for loops in bash

### ISSUE THREAD

**Issue by e-gillies-ix** [conda/conda#3296]
Hello Conda Devs!  

Great package overall, just a potential issue I've spotted in my own usage.  Basically, the IFS variable is set in whenever `source activate <env>` is called.  I think the error occurs here, [in line 95 the activation script on the current head.](https://github.com/conda/conda/blob/97248c5a8b5c84c35415530fd6d7258e2050a2cd/shell/activate#L95)

This causes whitespace delimination to no longer work in bash `for` loops.  I have not checked the other shells.  

To recreate, verify that before any activation is done, the following command has the default behaviour in bash : 

``` bash
bash$ for p in `echo hello:there:you:bug | sed 's/:/ /g'`; do echo $p; done
hello
there
you
bug
```

To trigger the bug, try creating a minimal environment with conda (python is used here, but any package should work), and insert an extra activation script in the correct directory: 

``` bash
bash$ conda create -n foo python=2.7
Fetching package metadata .......
Solving package specifications: ..........
Using Anaconda Cloud api site https://api.anaconda.org

Package plan for installation in environment <install_prefix>/anaconda2/envs/foo:

The following NEW packages will be INSTALLED:

    openssl:    1.0.2h-1     
    pip:        8.1.2-py27_0 
    python:     2.7.12-1     
    readline:   6.2-2        
    setuptools: 25.1.6-py27_0
    sqlite:     3.13.0-0     
    tk:         8.5.18-0     
    wheel:      0.29.0-py27_0
    zlib:       1.2.8-3      

Proceed ([y]/n)? y

Linking packages ...
[      COMPLETE      ]|################################################################################################################################################################| 100%
#
# To activate this environment, use:
# $ source activate foo
#
# To deactivate this environment, use:
# $ source deactivate
#
```

Activate the environment, create the directory for the extra activation scripts, write a dummy shell script to be sourced at line 95 of the main activation script, hence triggering the IFS bug on next activation.

``` bash
bash$ source activate foo
(foo) bash$ mkdir -p ${CONDA_PREFIX}/etc/conda/activate.d
(foo) bash$ echo 'echo bar' > ${CONDA_PREFIX}/etc/conda/activate.d/fooActivate.sh
(foo) bash$ source deactivate
```

Reactivate the environment to trigger IFS related bug:

``` bash
bash$ source activate foo
bar
(foo) bash$ for p in `echo hello:there:you:bug | sed 's/:/ /g'`; do echo $p; done
hello there you bug 
```

So, now IFS is actually set to something, so whitespace delimination is now broken in for loops.  This is especially harmful to any script that loops through the PATH variable in the above fashion.

**Comment by e-gillies-ix** [comment]
Pretty sure that issue #3128 is related, but haven't check explicitly.

**Comment by e-gillies-ix** [comment]
I tried to do a "clever" fix that remembers any IFS settings before usage of (de)activation scripts, and resets them to these values once IFS is used in these scripts, but it seemed a but unstable.  See [this commit](https://github.com/conda/conda/pull/3297/commits/fdcb52f4df4f1aeb1d8cd1e1e6359d88aec147ed) for the "clever" version.

Otherwise, I've opened a new pull request (#3297) to fix this, simply unsetting IFS once it is used.

Perhaps I missed something, and this change is invalid.  If so, please correct my contribution and keep us posted.  Thanks!

**Comment by e-gillies-ix** [comment]
Should this merge request (#3297) be accepted, I now see the behaviour I expect : 
```bash
bash$ source activate foo
hello
(foo) bash$ for p in `echo hello:there:you:bug | sed 's/:/ /g'`; do echo $p; done
hello
there
you
bug
(foo) bash$ source deactivate
bash$ for p in `echo hello:there:you:bug | sed 's/:/ /g'`; do echo $p; done
hello
there
you
bug
```

**Comment by e-gillies-ix** [comment]
As discussed in the (now closed) pull request #3297, this should be fixed once  #3175 is merged in.

**Comment by mingwandroid** [comment]
This was fixed with https://github.com/conda/conda/commit/c4bd2279e68af495d7a3bd60c3a03b15cdbc2318
